components/mosart/src/cpl/rof_comp_esmf.F90

```diff
@@ -705,7 +705,7 @@ subroutine rof_import_esmf( x2r_array, rc)
        else
           rtmCTL%qdto(n,nliq) = 0.0_r8
        endif
-       rtmCTL%qdem(n,nliq) = fptr(index_x2r_Flrl_demand,n2) * (rtmCTL%area(n)*0.001_r8)
+       rtmCTL%qdem(n,nliq) = fptr(index_x2r_Flrl_demand,n2) / TUnit%domainfrac(n) * (rtmCTL%area(n)*0.001_r8)
```

Comment:
It is better to make sure TUnit%domainfrac > 0.

Reply:
Added check when read in the data